src/gluonts/evaluation/_base.py

```diff
@@ -310,7 +310,7 @@ def extract_past_data(
         # Remove the prediction range
         # If the prediction range is not in the end of the time series,
         # everything after the prediction range is truncated
-        date_before_forecast = forecast.index[0] - forecast.index[0].freq
+        date_before_forecast = forecast.index[0] - forecast.start_date.freq
```

Comment:
@lostella this change is needed because irregular datetime index does not have a freq, thus I need to use the freq assigned to the start date